Codensity/lift

```diff
@@ -0,0 +1,3 @@
+  λ(f : Type → Type)
+→ λ(comonad : ./../Comonad/Type f)
```

Comment:
Maybe I'm wrong, but shouldn't this be `Monad`?

Reply:
I don't really understand the theory behind it, but there seems to be some duality to the `lift`/`lower` functions.

* When `lift`ing to a Kan-extension, we need a way to extract the value.
* When `lower`ing from a Kan-extension, we need a way to embed the value.

It's very possible that I've encoded the extensions incorrectly. Is there something we can check?

Reply:
Do you need have any literature where I can read up on it? But I was under the impression that `Codensity/lift` === `Transformer/lift` and the `Codensity` deals with `Monad`s and `Density` deals with `Comonad`s.

Reply:
Ah! Right.

To define `g a -> Ran f g a`, we need a function `g (f b) -> g b`, for any `b`. We can make that if we have `f b -> b` for any `b` and that `g` is a `Functor`. If `f = g` (as in `Codensity f a`), then we need a function `f (f b) -> f b`, for any `b`. That's monadic `join`.

When we specialize the right Kan-extension of `f` along itself, the problem can be solved in a different way.

Would you prefer to go with `g (f b) -> g b` instead of `Comonad/Type f` for `Ran/lift`? It doesn't make `f a -> Yoneda f a` any harder (as we need only supply identity), and would keep `Codensity` talking about `Monad`s.

Reply:
I guess this is one step removed from the similar [toRan][]. Maybe we go for that instead? It makes effectively no difference to `Codensity` or `Yoneda`, and gives the universal property for `Ran`. Seems like a win-win.

[toRan]: https://hackage.haskell.org/package/kan-extensions-5.2/docs/Data-Functor-Kan-Ran.html#v:toRan

Reply:
@FintanH Pushed some changes based on this idea. Can you give it a once over and see if this is more in line with what you're thinking?